backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/DocumentationUnitService.java

```diff
@@ -520,6 +520,35 @@ public RisJsonPatch updateDocumentationUnit(
     return toFrontend;
   }
 
+  /** Return a documentation unit with a new version */
+  private static DocumentationUnit updateDocumentationUnitVersion(
+      DocumentationUnit documentationUnit, long newVersion) {
+    if (documentationUnit instanceof Decision docUnit) {
+      documentationUnit = docUnit.toBuilder().version(newVersion).build();
+
+    } else if (documentationUnit instanceof PendingProceeding pendingProceeding) {
+      documentationUnit = pendingProceeding.toBuilder().version(newVersion).build();
+    }
+    return documentationUnit;
+  }
+
+  private DocumentationUnit updateDocumentationUnit(
+      User user,
+      DocumentationUnit patchedDocumentationUnit,
+      DuplicateCheckStatus duplicateCheckStatus)
+      throws DocumentationUnitNotExistsException {
+    DocumentationUnit updatedDocumentationUnit;
+    if (patchedDocumentationUnit instanceof Decision docUnit) {
+      updatedDocumentationUnit = updateDocumentationUnit(docUnit, duplicateCheckStatus, user);
+    } else if (patchedDocumentationUnit instanceof PendingProceeding pendingProceeding) {
+      updatedDocumentationUnit = updatePendingProceeding(pendingProceeding, user);
+    } else {
+      throw new DocumentationUnitException(
+          "Update not supported for Documentable type: " + patchedDocumentationUnit.getClass());
+    }
```

Comment:
Maybe we should throw a similar exception in `updateDocumentationUnitVersion` above?